L1Trigger/TrackFindingTracklet/src/TrackletParametersMemory.cc

```diff
@@ -34,17 +35,26 @@ void TrackletParametersMemory::writeMatches(Globals* globals, int& matchesL1, in
 }
 
 void TrackletParametersMemory::writeTPAR(bool first) {
+  const string dirTP = settings_.memPath() + "TrackletParameters/";
+
   std::ostringstream oss;
-  oss << "../data/MemPrints/TrackletParameters/TrackletParameters_" << getName() << "_" << std::setfill('0')
-      << std::setw(2) << (iSector_ + 1) << ".dat";
+  oss << dirTP << "TrackletParameters_" << getName() << "_" << std::setfill('0') << std::setw(2) << (iSector_ + 1)
+      << ".dat";
   auto const& fname = oss.str();
 
   if (first) {
     bx_ = 0;
     event_ = 1;
-    out_.open(fname.c_str());
+
+    if (not std::filesystem::exists(dirTP)) {
+      system((string("mkdir -p ") + dirTP).c_str());
```

Comment:
check if successful

Reply:
DONE
--  P.S. As this file/directory creation code appears in dozens of places, a future PR (almost ready) will move it to a single common function, to reduce duplicate code.